Sources/Penny/SOChecker.swift

```diff
@@ -0,0 +1,147 @@
+import Logging
+import DiscordBM
+import Markdown
+import Foundation
+
+actor SOChecker {
+
+    struct Storage: Sendable, Codable {
+        var lastCheckDate: Date? = nil
+    }
+
+    var storage = Storage()
+
+    let soService: any SOService
```

Comment:
It would be much preferable to use `some SOService` rather than `any SOService` if possible (not really relevant to performance in this case, but as a general rule `some` is always better than `any` if it's valid at all).

Reply:
Not possible because we can't mock it if we use `some`.